### PR TITLE
INS-2892: Move ExecutionBroker from ExecutionState to CurrentExecution

### PR DESCRIPTION
**Pull request by bigbes**

### CODE REVIEW

File: logicrunner/logicrunner_unit_test.go [hunk 1]
```diff
@@ -102,7 +102,7 @@ func (suite *LogicRunnerCommonTestSuite) SetupLogicRunner() {
 }
 
 func (suite *LogicRunnerCommonTestSuite) AfterTest(suiteName, testName string) {
-	suite.mc.Wait(2 * time.Minute)
+	suite.mc.Wait(2 * time.Second)
```
Comment (ruz) [inline logicrunner/logicrunner_unit_test.go]: this again

Comment (bigbes) [reply logicrunner/logicrunner_unit_test.go]: Fixed

File: logicrunner/logicrunner.go [hunk 2]
```diff
@@ -57,8 +57,20 @@ type Ref = insolar.Reference
...
 	sync.Mutex
 
-	ExecutionState *ExecutionState
-	Validation     *ExecutionState
+	ExecutionBroker *ExecutionBroker
+	ExecutionState  *ExecutionState
+	Validation      *ExecutionState
+}
+
+func (st *ObjectState) InitAndGetExecution(lr *LogicRunner, ref *insolar.Reference) (*ExecutionState, *ExecutionBroker) {
```
Comment (ruz) [inline logicrunner/logicrunner.go]: this is wrong. please make it "UpsertExecution" (or something like that) on StateStorage as it interface. Inject LR or anything else required to complete operation as dependency of StateStorage, don't pass it as argument into method.

File: server/internal/virtual/components.go [hunk 3]
```diff
@@ -154,8 +153,8 @@ func initComponents(
 	checkError(ctx, err, "failed to load VersionManager: ")
 
 	// move to logic runner ??
-	err = logicRunner.OnPulse(ctx, *pulsar.NewPulse(cfg.Pulsar.NumberDelta, 0, &entropygenerator.StandardEntropyGenerator{}))
-	checkError(ctx, err, "failed init pulse for LogicRunner")
+	// err = logicRunner.OnPulse(ctx, *pulsar.NewPulse(cfg.Pulsar.NumberDelta, 0, &entropygenerator.StandardEntropyGenerator{}))
+	// checkError(ctx, err, "failed init pulse for LogicRunner")
```
Comment (ruz) [inline server/internal/virtual/components.go]: delete this code

File: logicrunner/logicrunner_unit_test.go [hunk 4]
```diff
@@ -99,10 +99,12 @@ func (suite *LogicRunnerCommonTestSuite) SetupLogicRunner() {
 	suite.lr.PulseAccessor = suite.ps
 	suite.lr.NodeNetwork = suite.nn
 	suite.lr.RequestsExecutor = suite.re
+
+	_ = suite.lr.Init(suite.ctx)
 }
 
 func (suite *LogicRunnerCommonTestSuite) AfterTest(suiteName, testName string) {
-	suite.mc.Wait(2 * time.Minute)
+	suite.mc.Wait(2 * time.Second)
```
Comment (ruz) [inline logicrunner/logicrunner_unit_test.go]: return back minutes!

File: logicrunner/logicrunner.go [hunk 5]
```diff
@@ -409,30 +376,28 @@ func (lr *LogicRunner) OnPulse(ctx context.Context, pulse insolar.Pulse) error {
...
 		}
 
-		if state.ExecutionState == nil && state.Validation == nil {
+		state.Unlock()
+
+		if state.Validation == nil && state.ExecutionState == nil {
 			lr.StateStorage.DeleteObjectState(ref)
 		}
-
-		state.Unlock()
```
Comment (ruz) [inline logicrunner/logicrunner.go]: this change is invalid! if condition  ```state.Validation == nil && state.ExecutionState == nil``` should be checked under lock.

Comment (bigbes) [reply logicrunner/logicrunner.go]: fixed

**Pull request merged**